src/main/java/core/basesyntax/service/impl/PurchaseAction.java

```diff
@@ -0,0 +1,10 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.service.ActionHandler;
+
+public class PurchaseAction implements ActionHandler {
```

Comment:
The apply method should have a return type of void only if it is guaranteed not to fail under any circumstance, which is not the case here. It would be better to return a boolean indicating success or throw an exception in case of failure.